commons/commons-lang/src/main/java/com/clicktravel/common/random/Randoms.java

```diff
@@ -229,6 +234,101 @@ public static BigDecimal randomPositiveBigDecimal(final int maxValue, final int
     }
 
     /**
+     * Return an int in the range <code>from</code> to <code>to</code> inclusive of from but exclusive of to e.g.
+     * randomIntInRange(1,20) could return an int from 1 to 19 inclusive. Negative numbers are also supported by this
+     * method e.g. randomIntInRange(-10,10) could return an int from -10 to 9 inclusive
+     *
+     * @param from the start of the range (inclusive)
+     * @param to the end of the range (exclusive)
+     *
+     * @return a random int
+     */
+    public static int randomIntInRange(final int from, final int to) {
+        if (from > to) {
+            throw new ValidationException(String.format("From (%d) must be less than to (%d)", from, to), "from", "to");
+        } else if (from == to) {
+            throw new ValidationException(String.format("From (%d) must be different to (%d)", from, to), "from", "to");
+        }
+        return from + Randoms.randomInt(to - from);
+    }
+
+    /**
+     * Generates a random valid credit card number.
+     *
+     * Code taken from a public Gist created by Josef Galea https://gist.github.com/josefeg/5781824
+     *
+     * @param prefix, used to identify the bank that is issuing the credit card, this should be a positive number
+     *
+     * @return A randomly generated, valid, credit card number.
+     */
+    public static String randomCreditCardNumber(final String prefix) {
+
+        Check.matchesPattern("prefix", prefix, CARD_PREFIX_PATTERN);
```

Comment:
This will throw a ```ValidationException``` in case of failure, which is not appropriate. Better to throw an ```IllegalArgumentException```:
```java
if (!ALL_DIGITS_PATTERN.matcher(prefix).matches()) { throw new IllegalArgumentException("Invalid card prefix"); }
```